gax/src/test/java/com/google/api/gax/batching/v2/BatchingCallSettingsTest.java

```diff
@@ -0,0 +1,145 @@
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.batching.v2;
+
+import static com.google.api.gax.rpc.testing.FakeBatchableApi.SQUARER_BATCHING_DESC_V2;
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.gax.retrying.RetrySettings;
+import com.google.api.gax.rpc.StatusCode;
+import com.google.api.gax.rpc.testing.FakeBatchableApi.LabeledIntList;
+import com.google.common.collect.Sets;
+import java.util.List;
+import java.util.Set;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.threeten.bp.Duration;
+
+@RunWith(JUnit4.class)
+public class BatchingCallSettingsTest {
+
+  private final BatchingSettings batchingSettings =
+      BatchingSettings.newBuilder()
+          .setElementCountThreshold(10)
+          .setRequestByteThreshold(20L)
+          .setDelayThreshold(Duration.ofMillis(5))
+          .build();
+  private BatchingCallSettings.Builder<Integer, Integer, LabeledIntList, List<Integer>> builder;
+
+  @Test
+  public void testEmptyBuilder() {
+    builder = BatchingCallSettings.newBuilder(SQUARER_BATCHING_DESC_V2);
+    assertThat(builder.getBatchingDescriptor()).isSameInstanceAs(SQUARER_BATCHING_DESC_V2);
+    assertThat(builder.getBatchingSettings()).isNull();
+    assertThat(builder.getRetryableCodes()).isEmpty();
+    assertThat(builder.getRetrySettings()).isNotNull();
+
+    builder.setBatchingSettings(batchingSettings);
+    BatchingCallSettings settings = builder.build();
+    assertSettings(settings);
+    assertThat(settings.getRetryableCodes()).isEmpty();
+    assertThat(settings.getRetrySettings()).isNotNull();
+  }
+
+  @Test
+  public void testBuilder() {
+    builder = BatchingCallSettings.newBuilder(SQUARER_BATCHING_DESC_V2);
+
+    Set<StatusCode.Code> retryCodes = Sets.newHashSet(StatusCode.Code.UNAVAILABLE);
+    RetrySettings retrySettings = RetrySettings.newBuilder().build();
+    builder.setBatchingSettings(batchingSettings);
+    builder.setRetryableCodes(retryCodes);
+    builder.setRetrySettings(retrySettings);
```

Comment:
I think it would be nicer to chain these:
```
    builder
       .setBatchingSettings(batchingSettings)
       .setRetryableCodes(retryCodes)
       .setRetrySettings(retrySettings);
```